core/src/config.rs

```diff
@@ -94,38 +96,45 @@ impl<WB: WasmBackend> MModuleConfig<WB> {
     }
 }
 
-pub struct MarineCoreConfig {
+pub struct MarineCoreConfig<WB: WasmBackend> {
     pub(crate) total_memory_limit: u64,
-}
-
-impl Default for MarineCoreConfig {
-    fn default() -> Self {
-        Self {
-            total_memory_limit: INFINITE_MEMORY_LIMIT,
-        }
-    }
+    pub(crate) wasm_backend: WB,
 }
 
 pub const INFINITE_MEMORY_LIMIT: u64 = u64::MAX;
 
 #[derive(Default, Debug)]
-pub struct MarineCoreConfigBuilder {
+pub struct MarineCoreConfigBuilder<WB: WasmBackend> {
     total_memory_limit: Option<u64>,
+    wasm_backend: Option<WB>,
 }
 
-impl MarineCoreConfigBuilder {
+impl<WB: WasmBackend> MarineCoreConfigBuilder<WB> {
     pub fn new() -> Self {
-        Self::default()
+        Self {
+            total_memory_limit: None,
+            wasm_backend: None,
+        }
     }
 
     pub fn total_memory_limit(mut self, total_memory_limit: u64) -> Self {
         self.total_memory_limit = Some(total_memory_limit);
         self
     }
 
-    pub fn build(self) -> MarineCoreConfig {
-        MarineCoreConfig {
+    pub fn wasm_backend(mut self, wasm_backend: WB) -> Self {
+        self.wasm_backend = Some(wasm_backend);
+        self
+    }
+
+    pub fn build(self) -> MResult<MarineCoreConfig<WB>> {
+        let config = MarineCoreConfig {
             total_memory_limit: self.total_memory_limit.unwrap_or(INFINITE_MEMORY_LIMIT),
-        }
+            wasm_backend: self
+                .wasm_backend
+                .unwrap_or(<WB as WasmBackend>::new_async()?),
```

Comment:
Maybe reqiure a WasmBackend argument in `MarineCoreConfigBuilder::new` instead? Or smth like this

Reply:
Yep, it looks better

Reply:
I proceeded and completely removed MarineConfigBuilder and kept only MarineConfig with a constructor. Feels like we still do not need the builder